src/Aquifer.Common/Services/Caching/CachingVersificationService.cs

```diff
@@ -0,0 +1,42 @@
+using Aquifer.Data;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Aquifer.Common.Services.Caching;
+
+public interface ICachingVersificationService
+{
+    Task<Dictionary<(int bibleVerseId, char? bibleVersePart), (int baseVerseId, char? baseVersePart)>> GetVersificationsByBibleIdAsync(
+        int bibleId, CancellationToken cancellationToken);
+}
+
+public class CachingVersificationService(AquiferDbContext _dbContext, IMemoryCache _memoryCache) : ICachingVersificationService
+{
+    private const string VersificationDictionariesCacheKey = $"{nameof(CachingVersificationService)}:VersificationDictionaries";
+    private static readonly TimeSpan s_cacheLifetime = TimeSpan.FromMinutes(30);
+
+    public async Task<Dictionary<(int bibleVerseId, char? bibleVersePart), (int baseVerseId, char? baseVersePart)>>
+        GetVersificationsByBibleIdAsync(int bibleId, CancellationToken cancellationToken)
+    {
+        var dictionaries = await GetDictionariesFromCacheAsync(cancellationToken);
+
+        var versificationDictionary = dictionaries?.GetValueOrDefault(bibleId) ?? throw new KeyNotFoundException();
```

Comment:
If this code is assuming that all Bibles will have versification data in the DB (a reasonable assumption) then I suggest throwing an `InvalidOperationException()` with a good error message stating that assumption.

Reply:
So far, I've been under the impression that it's not assuming that there will be versification for every Bible

Reply:
Definitely wouldn't assume they'll all have it.

Reply:
Seems like we'd just return an empty dictionary here rather than blow up.